corehq/warehouse/transforms/sql/application_dim.sql

```diff
@@ -25,3 +25,12 @@ SELECT
     now(),
     '{{ batch_id }}'
 FROM {{ application_staging }}
+ON CONFLICT (application_id) DO UPDATE
+SET application_id = EXCLUDED.application_id
+    domain = EXCLUDED.domain,
+    name = EXCLUDED.name,
+    application_last_modified = EXCLUDED.application_last_modified,
+    deleted = EXCLUDED.deleted,
```

Comment:
does referencing `EXCLUDED.deleted` work? if so then we're ok, but if not we may need to alias the column in the `SELECT` query. i'm not familiar with the upsert syntax so i don't know if it's using the column names from the select or from the insert

Reply:
my understanding is that this should work. from the pg documentation
> The SET and WHERE clauses in ON CONFLICT DO UPDATE have access to the existing row using the table's name (or an alias), and to rows proposed for insertion using the special excluded table.

I interpreted deleted as part of the "row proposed for insertion", but let me know if that does not seem right to you. I also played around with sql fiddle and it appeared to work.
https://www.postgresql.org/docs/9.6/static/sql-insert.html